File: Easy/PalindromeNumber.java
/*
Determine whether an integer is a palindrome. Do this without extra space.

Some hints:
Could negative integers be palindromes? (ie, -1)

If you are thinking of converting the integer to string, note the restriction of using extra space.

You could also try reversing an integer. However, if you have solved the problem "Reverse Integer", you know that the reversed integer might overflow. How would you handle such case?

There is a more generic way of solving this problem.
*/

public class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) return false;
        int z = x;
        long y = 0;
        while (z != 0){
        	y = 10*y + z%10;
        	z = z/10;
        }
        return x == y;
    }
}

File: Easy/PalindromeNumber.py
"""
Determine whether an integer is a palindrome. Do this without extra space.

Some hints:
Could negative integers be palindromes? (ie, -1)

If you are thinking of converting the integer to string, note the restriction of using extra space.

You could also try reversing an integer. However, if you have solved the problem "Reverse Integer", you know that the reversed integer might overflow. How would you handle such case?

There is a more generic way of solving this problem.
"""

class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if (x < 0):
        	return False  // assume that negative numbers are not palindrome
        z = x
        y = 0
        while (z != 0):
        	y = 10*y + z%10
        	z = z/10
        return x == y
